modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C

```diff
@@ -0,0 +1,496 @@
+  std::string region_id_name = "region_id";
+  std::string pin_type_id_name = "pin_type_id";
+
+  if (!(*_build_mesh)->has_elem_integer(region_id_name) ||
+      !(*_build_mesh)->has_elem_integer(pin_type_id_name))
+    mooseError("Expected mesh inputs to have extra integer ids");
+
+  unsigned int ptid_int = (*_build_mesh)->get_elem_integer_index(pin_type_id_name);
+  unsigned int rid_int = (*_build_mesh)->get_elem_integer_index(region_id_name);
+
+  // go through the background and duct regions and assign their elements
+  // pin_type id placeholders and region id place holders
+  for (size_t i = 0; i < _peripheral_regions; ++i)
+  {
+    subdomain_id_type region = UINT16_MAX - (i + 1);
+
+    for (const auto & elem : as_range((*_build_mesh)->active_subdomain_elements_begin(region),
+                                      (*_build_mesh)->active_subdomain_elements_end(region)))
+    {
+      elem->set_extra_integer(ptid_int, region); // identifier for element not in a pin
+      elem->set_extra_integer(rid_int, region);
+    }
+  }
+
+  if (_extrude)
+  {
+    // swap the region ids on the subdomain ids to the correct ones
+    // for their axial layer
+    mesh_name = name() + "_extrudedIDs";
+    if (!(*_build_mesh)->has_elem_integer(plane_id_name))
+      mooseError("Expected extruded mesh to have plane IDs");
+
+    unsigned int pid_int = (*_build_mesh)->get_elem_integer_index(plane_id_name);
+
+    for (auto & elem : (*_build_mesh)->active_element_ptr_range())
+    {
+      dof_id_type z_id = elem->get_extra_integer(pid_int);
+      dof_id_type pt_id = elem->get_extra_integer(ptid_int);
+      subdomain_id_type r_id = elem->subdomain_id();
+
+      // Going through the elements in the mesh checking pin_type_ids to assign their axial
+      // region ids
+      if (_id_map.find(pt_id) == _id_map.end())
+      {
+        // region isn't in a pin so it must be a peripheral (background or duct) region of the
+        // assembly
+        unsigned int peripheral_index = (UINT16_MAX - 1) - r_id;
+        if (peripheral_index == 0)
+          // background region element
+          elem->subdomain_id() = _background_region_id[z_id];
+        else
+          // duct region element
+          elem->subdomain_id() = _duct_region_ids[z_id][peripheral_index - 1];
+      }
+      else
+      {
+        // region is in a pin so grab the different axial region ids and swap them
+        // since during extrusion all regions are given the same ID as the 2D layer
+        for (size_t i = 0; i < (_id_map.at(pt_id))[0].size(); ++i)
+        {
+          // swap subdomain region ids if they are different
+          if (r_id == _id_map.at(pt_id)[0][i] &&
+              _id_map.at(pt_id)[0][i] != _id_map.at(pt_id)[z_id][i])
+            elem->subdomain_id() = _id_map.at(pt_id)[z_id][i];
+        }
+      }
+    }
+  }
+  else
+  {
+    mesh_name = name() + "_pattern";
+
+    for (size_t i = 0; i < _peripheral_regions; ++i)
```

Comment:
We explicitly specify the std namespace everywhere, so this should be 
```suggestion
    for (std::size_t i = 0; i < _peripheral_regions; ++i)
```

Reply:
Alternatively you can youse this, which is what we've lately been transitioning to

```suggestion
    for (const auto i: make_range(_peripheral_regions))
```

(requires `#include "libmesh/int_range.h"`

Reply:
Thanks, I've gone ahead and updated most for loops to incorporate this new convention.